src/main/java/com/github/orest/car_shop/controllers/CarApiController.java

```diff
@@ -0,0 +1,72 @@
+package com.github.orest.car_shop.controllers;
+
+import com.github.orest.car_shop.exceptions.CarNotFoundException;
+import com.github.orest.car_shop.model.Car;
+import com.github.orest.car_shop.repository.CarRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.servlet.view.RedirectView;
+
+import javax.validation.Valid;
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+@Controller
+@RestController
```

Comment:
You don’t need to use @Controller and @RestController in one place. Please open the source code of @Controller and you will see why